providers/oauth/oauth.webrequest.js

```diff
@@ -42,26 +42,29 @@ ChromeWebRequestAuth.prototype.initiateOAuth = function(redirectURIs, continuati
   return false;
 };
 
-ChromeWebRequestAuth.prototype.launchAuthFlow = function(authUrl, stateObj, continuation) {
+ChromeWebRequestAuth.prototype.launchAuthFlow = function(authUrl, stateObj, interactive, continuation) {
   "use strict";
+  if (interactive === undefined) {
+    interactive = true;
+  }
   var listener = this.reqListener.bind(this, stateObj, continuation);
   this.listeners[stateObj.state] = listener;
   chrome.webRequest.onBeforeRequest.addListener(listener, {
     types: ["main_frame"],
     urls: [stateObj.redirect]
   });
-  
+
   chrome.tabs.create({
     url: authUrl,
-    active: true
+    active: interactive
```

Comment:
Not exactly.  A bit of backstory - all of these changes are being done to make Facebook reconnect a better UX.  Facebook, unlike Google, doesn't let us get refresh (long-lived) tokens, so there is no way for us to get a valid Facebook token without opening an oauth tab.  However, in most cases where we need to reconnect a user to Facebook, the user is going to be still logged into Facebook in their browser and still granting uProxy permission - this means that the auth tab will just open, redirect to uproxy.org with the token, then uProxy will close the tab.  It is fairly annoying right now when this happens and the tab is focused on (a common scenario for me is I'm logged into FB on uProxy, lock my computer, come back and start doing unrelated stuff in Chrome, then suddenly a tab opens and closes).

What we want for uProxy is for this tab to open in the background (not be focused on).  We've had this briefly working in Chrome where we override the core.oauth provider.

In practice this code in freedom-for-chrome won't actually be used in uProxy.  Instead uProxy will use it's own core.oauth provider for Chrome, and will use the default oauth provider for Firefox where I am making a similar change to not select the oauth tab if !interactive.